include/alpaka/test/Check.hpp

```diff
@@ -9,28 +9,46 @@
 
 #pragma once
 
+#include <alpaka/core/BoostPredef.hpp>
+
 #include <cstdio>
 
 // TODO: SYCL doesn't have a way to detect if we're looking at device or host code. This needs a workaround so that
 // SYCL and other back-ends are compatible.
 #ifdef ALPAKA_ACC_SYCL_ENABLED
-#    define ALPAKA_CHECK(success, expression)                                                                         \
+#    define ALPAKA_CHECK_DO(printMsg, file, line, success, expression)                                                \
         do                                                                                                            \
         {                                                                                                             \
             if(!(expression))                                                                                         \
             {                                                                                                         \
-                acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")'\n";                               \
+                acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")' in " << file << ":" << line       \
+                         << "\n";                                                                                     \
```

Comment:
```suggestion
                acc.cout << "ALPAKA_CHECK failed because '!(" << #expression << ")' in " << __FILE__<< ":" << __LINE__\
                         << "\n";                                                                                     \
```
You should be able to use the file and line macros directly inside the macro definition and not pass them around.